File: src/main/java/br/com/leonardoferreira/jirareport/domain/request/UpdateBoardRequest.java
package br.com.leonardoferreira.jirareport.domain.request;

import br.com.leonardoferreira.jirareport.domain.ImpedimentType;
import br.com.leonardoferreira.jirareport.domain.vo.DynamicFieldConfig;
import java.util.List;
import lombok.Data;

@Data
public class UpdateBoardRequest {

    private String name;

    private String startColumn;

    private String endColumn;

    private List<String> fluxColumn;

    private List<String> ignoreIssueType;

    private String epicCF;

    private String estimateCF;

    private String systemCF;

    private String projectCF;

    private Boolean calcDueDate;

    private Boolean ignoreWeekend;

    private ImpedimentType impedimentType;

    private List<String> impedimentColumns;

    private List<DynamicFieldConfig> dynamicFields;

    private List<String> touchingColumns;

    private List<String> waitingColumns;

}

File: common-library/src/main/kotlin/br/com/jiratorio/domain/request/UpdateBoardRequest.kt
package br.com.jiratorio.domain.request

import br.com.jiratorio.domain.DueDateType
import br.com.jiratorio.domain.ImpedimentType
import br.com.jiratorio.domain.issueperiodnameformat.IssuePeriodNameFormat
import javax.validation.constraints.NotBlank

data class UpdateBoardRequest(

    @field:NotBlank
    val name: String,

    val startColumn: String? = null,

    val endColumn: String? = null,

    val fluxColumn: MutableList<String>? = null,

    val ignoreIssueType: MutableList<String>? = null,

    val epicCF: String? = null,

    val estimateCF: String? = null,

    val systemCF: String? = null,

    val projectCF: String? = null,

    val ignoreWeekend: Boolean? = null,

    val impedimentType: ImpedimentType? = null,

    val impedimentColumns: MutableList<String>? = null,

    val touchingColumns: MutableList<String>? = null,

    val waitingColumns: MutableList<String>? = null,

    val dueDateCF: String? = null,

    val dueDateType: DueDateType? = null,

    val useLastOccurrenceWhenCalculateLeadTime: Boolean = false,

    val issuePeriodNameFormat: IssuePeriodNameFormat = IssuePeriodNameFormat.INITIAL_AND_FINAL_DATE

)
